drivers/serial/CMakeLists.txt

```diff
@@ -28,6 +28,7 @@ zephyr_library_sources_ifdef(CONFIG_UART_SAM uart_sam.c)
 zephyr_library_sources_ifdef(CONFIG_USART_SAM usart_sam.c)
 zephyr_library_sources_ifdef(CONFIG_UART_STELLARIS uart_stellaris.c)
 zephyr_library_sources_ifdef(CONFIG_UART_STM32 uart_stm32.c)
+zephyr_library_sources_ifdef(CONFIG_UART_STM32 uart_bm.c)
```

Comment:
Worth thinking about renaming this series of files to uart_stm32_bm, to both show that parts of it originated out of the uart_stm32 driver and also that it only works on STM32.